react-router/src/Router.md

```diff
@@ -1,115 +1,193 @@
+### 5. Dynamic Routing and Parameters
+React Router supports dynamic routing, allowing parameters to be passed within the URL, making it easier to build dynamic, data-driven applications.
 
-6. **Essential Components**:
-   - **BrowserRouter**: Sets up the application's navigation context.
-   - **Route**: Defines individual routes and associates them with specific components.
-   - **Switch**: Ensures that only one route is rendered at a time.
+---
 
-### Setting up the project
+## Essential Components of React Router
 
-1. **Create a react project using vite bundler** - 
-    - Create the project using the command ` npm create vite@latest `
-    - Enter in the project directory by using the command ` cd <project-name> `
-    - install the node module package using the command ` npm install `
-    - To start the localhost server use the command ` npm run dev `
+- **`<BrowserRouter>`**: The root component that sets up the routing context.
+- **`<Routes>`**: A wrapper for all the route definitions.
+- **`<Route>`**: Defines the path and the component that should render.
+- **`<NavLink>`**: Provides navigation links to different routes.
+- **`<Outlet>`**: Used to render child routes in a parent route.
 
-2. ** Follow the steps to remove conetent ** -
-    - Remove the pre-written code from the ` App.jsx ` file
-    - Your `App.js` should look like this
+---
 
- ```jsx
-import React from "react";
-import "./index.css"
+## Setting Up the Project
 
-export default function App() {
-  return (
-    <div>
-        <h1>React Router</h1>
-    </div>
-  )
+### 1. Create a React Project using Vite
+
+```bash
+npm create vite@latest
+cd <project-name>
+npm install
+npm run dev
+
+## 2. Install React Router
+
+To install React Router, use the following command:
+
+```bash
+npm install react-router-dom
 ```
+## 3. Basic Setup in `App.jsx`
 
-3. ** Install React-router ** - 
-    - open the terminal 
-    - Install the react-router library in your current project 
-    - Uisng the command ` npm install react-router-dom `
+In this basic setup, we will define three pages: **Home**, **About**, and **Contact**. We’ll also implement navigation using `NavLink`:
 
-4. ** Add the code to `App.jsx` file ** - 
-    - Update the existing code to the following code in ` App.jsx ` file inside the ` src ` folder
 ```jsx
 import React from "react";
-import "./index.css"
+import { BrowserRouter as Router, Route, Routes, NavLink } from "react-router-dom";
+import Home from "./components/Home";
+import About from "./components/About";
+import Contact from "./components/Contact";
 
-export default function App() {
+function App() {
   return (
-    <main>
+    <Router>
       <nav>
-        <ul>
-          <li><a href="/">Home</a></li>
-          <li><a href="/about">About</a></li>
-          <li><a href="/contact">Contact</a></li>
-        </ul>
-        </nav>
-     </main>
+        <NavLink to="/" exact>Home</NavLink>
```

Comment:
Please explain the code for the better understanding such as `NavLink` , `Route` , `Routes` , `path` , `Element` , `Router`